pkg/operator/resources/job/batchapi/job_status.go

```diff
@@ -34,22 +39,57 @@ func GetJobStatus(jobKey spec.JobKey) (*status.BatchJobStatus, error) {
 		return nil, err
 	}
 
-	k8sJob, err := config.K8s.GetJob(jobKey.K8sName())
-	if err != nil {
-		return nil, err
-	}
-
-	pods, err := config.K8s.ListPodsByLabels(map[string]string{"apiName": jobKey.APIName, "jobID": jobKey.ID})
-	if err != nil {
+	ctx := context.Background()
+	var batchJob batch.BatchJob
+	if err = config.K8s.Get(ctx, client.ObjectKey{Name: jobKey.ID, Namespace: config.K8s.Namespace}, &batchJob); err != nil {
 		return nil, err
 	}
 
-	return getJobStatusFromJobState(jobState, k8sJob, pods)
+	return getJobStatusFromJobState(jobState, &batchJob)
 }
 
-func getJobStatusFromJobState(jobState *job.State, k8sJob *kbatch.Job, pods []kcore.Pod) (*status.BatchJobStatus, error) {
+func getJobStatusFromJobState(jobState *job.State, batchJob *batch.BatchJob) (*status.BatchJobStatus, error) {
 	jobKey := jobState.JobKey
 
+	if batchJob != nil {
+		if batchJob.Status.Status == status.JobUnknown || batchJob.Status.Status.IsNotStarted() {
```

Comment:
It looks like we can make things easier by changing the order of the enum so that the 0th index is pending as discussed on call

Reply:
I am a bit afraid of doing this now because of TaskAPI